src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,40 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login must be at least "
+                    + MIN_LOGIN_LENGTH + " characters long");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password must be at least "
+                    + MIN_PASSWORD_LENGTH + " characters long");
+        }
+
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Not valid age " + user.getAge()
+                    + ". Min acceptable age is " + MIN_AGE);
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User is already exist");
+        }
+        storageDao.add(user);
+        return user;
```

Comment:
After adding the user to storage, it would be good to validate that the user has been successfully added by checking the return value of the storageDao.add method.